tests/test_salesforce_api_per_run_quota.py

```diff
@@ -0,0 +1,53 @@
+"""
+Test that with only custom fields selected for a stream automatic fields and custom fields  are still replicated
+"""
+import copy
+from tap_tester.base_suite_tests.api_quota_test import APIQuotaTest
+from sfbase import SFBaseTest
+from tap_tester import runner, menagerie
+
+
+class SFAPIQuota(APIQuotaTest, SFBaseTest):
+    """Test that with only custom fields selected for a stream automatic fields and custom fields are still replicated
+    TODO
+    This test coveres only BULK api and just couple of streams. We have separate cards to cover these cases
+    TDL-23653: [tap-salesforce]: QA - Add all the streams for the all_fields test
+    TDL-23654: [tap-salesforce]: QA - Add all-fields testcase for REST API streams
+    """
+
+    salesforce_api = 'BULK'
+    expected_error = "Terminating replication due to allotted quota of 1.0% per replication"
+
+    @staticmethod
+    def name():
+        return "tt_sf_api_quota"
+
+    def get_properties(self, original: bool = True):
+        """Configuration properties required for the tap."""
+        return_value = {
+            'start_date': '2000-11-23T00:00:00Z',
+            'instance_url': 'https://singer2-dev-ed.my.salesforce.com',
+            'select_fields_by_default': 'true',
+            'quota_percent_total': '85',
+            'quota_percent_per_run' : '1',
+            'api_type': self.salesforce_api,
+            'is_sandbox': 'false'
+        }
+        if original:
+            return return_value
+
+        # This test needs the new connections start date to be larger than the default
+        assert self.start_date > return_value["start_date"]
```

Comment:
What is the intent of this assertion?  I'm guessing that we just want to prevent start date from being too old.  If that is the case then I'd suggest setting start date dynamically.  If we are depending on what is specified in base, it will eventually have the potential to cause the same amount of issues as using the date defined in properties.

Reply:
Probably not needed, this method was overridden from sfbase.py